javascript/nodejs/exec_os_command.yaml

```diff
@@ -0,0 +1,74 @@
+rules:
+  - id: generic_os_command_exec
+    patterns:
+      - pattern-inside: |
+          var $EXEC = require('child_process');
+          ...
+          $APP.$METHOD(..., function $FUNC($REQ, $RES, ...){ ... });
+      - pattern-either:
+          - pattern: |
+              $EXEC.exec(..., $REQ.$QUERY.$VAR, ...)
+          - pattern: |
+              $EXEC.exec(..., "..." + $REQ.$QUERY.$VAR, ...)
+          - pattern: |
+              $EXEC.exec(..., $REQ.$QUERY.$VAR + "...", ...)
+          - pattern: |
+              $EXEC.exec(..., "..." + $REQ.$QUERY.$VAR + "...", ...)
+          - pattern: |
+              $EXEC.exec(..., $REQ.$QUERY, ...)
+          - pattern: |
+              $EXEC.exec(..., "..." + $REQ.$QUERY, ...)
+          - pattern: |
+              $EXEC.exec(..., $REQ.$QUERY + "...", ...);
+          - pattern: |
+              $EXEC.exec(..., "..." + $REQ.$QUERY + "...", ...)
+          - pattern: |
+              $EXEC.execSync(..., $REQ.$QUERY.$VAR, ...)
+          - pattern: |
+              $EXEC.execSync(..., "..." + $REQ.$QUERY.$VAR, ...)
+          - pattern: |
+              $EXEC.execSync(..., $REQ.$QUERY.$VAR + "...", ...)
+          - pattern: |
+              $EXEC.execSync(..., "..." + $REQ.$QUERY.$VAR + "...", ...)
+          - pattern: |
+              $EXEC.execSync(..., $REQ.$QUERY, ...)
+          - pattern: |
+              $EXEC.execSync(..., "..." + $REQ.$QUERY, ...)
+          - pattern: |
+              $EXEC.execSync(..., $REQ.$QUERY + "...", ...)
+          - pattern: |
+              $EXEC.execSync(..., "..." + $REQ.$QUERY + "...", ...)
+    message: >-
+      User controlled data in 'child_process.exec()' can result in Remote OS
+      Command Execution.
+    languages:
+      - js
+    severity: ERROR
+    metadata:
+      owasp: 'A1: Injection'
+      cwe: >-
+        CWE-78: Improper Neutralization of Special Elements used in an OS
+        Command ('OS Command Injection')
+  - id: node_os_command_exec
+    patterns:
+      - pattern-not-inside: |
+          $APP.$METHOD(..., function $FUNC($REQ, $RES, ...) { ... })
+      - pattern-not:
+          $X.exec("...", ...)
+      - pattern-not:
+          $X.execSync("...", ...)
+      - pattern: |
```

Comment:
@ajinabraham I think these need to be inside a `pattern-either` because things in `patterns` are ANDed together and so you'll never have `$X.execSync` and `$.exec`

Reply:
Good catch!